### PR TITLE
pkg/lua: Fix warnings on discarded qualifiers.

### PR DESCRIPTION
**Pull request by jcarrano**
#### Suppress warnings on the upstream code

Upstream lua has many instances of discarding cont qualifiers throughexplicit casting. This disables the warnigns for the package so that they don't show up when one enables -Wcast-qual globally, as I aspire to do in https://github.com/RIOT-OS/RIOT/issues/10177

####  Fix warning on my own code

There was a cast in a macro that calculated offsets. Since it was only doing a pointer difference, the cast was inocuous, but I fix it anyways.

### Testing procedure

Compile `examples/lua_REPL` with `WERROR=0 CFLAGS=-Wcast-qual make`

After these patches, lua and lua-contrib should report no more warnings of that type.

### Issues/PRs references

Small fix towards #10177 .

### CODE REVIEW

**Comment by jcarrano** [comment]
@danpetry Do you mind taking a look at this?

**Comment by tcschmidt** [comment]
@jcarrano ping.

**Comment by jcarrano** [comment]
@danpetry I fixed the comment, finally.

File: pkg/lua/Makefile.lua [hunk 1]
```diff
@@ -6,6 +6,9 @@ endif
 
 CFLAGS += -DLUA_MAXCAPTURES=16 -DL_MAXLENNUM=50
 
+# Upstream has several of this which will not be fixed
```
[the remark this answers is not in the document]
Comment (danpetry) [reply pkg/lua/Makefile.lua]: Great, thanks

**Pull request merged**